### PR TITLE
deprecate old-style v1 JSON filter config

### PR DESCRIPTION
**Pull request by mattklein123**
Now that TCP proxy has parity, we can completely remove v1
style config loading.

Risk Level: Low
Testing: New UT
Docs Changes: Added.
Release Notes: Added.

### CODE REVIEW

**Comment by mattklein123** [comment]
@alyssawilk @derekargueta PTAL

**Comment by derekargueta** [comment]
No code changes? Might be useful to users to hook this into the deprecated feature counter or start-up logging.

**Comment by derekargueta** [comment]
Also is this different than the v1 deprecation in the [1.8 deprecation notes](https://github.com/envoyproxy/envoy/blob/master/docs/root/intro/deprecated.rst#version-180-oct-4-2018) (first bullet)?

**Comment by alyssawilk** [comment]
@PiotrSikora for parity comments, since this relates to #5355

If we're going to do this we should flag the actual API fields as well so the scripts pick it up, no?

**Comment by mattklein123** [comment]
> If we're going to do this we should flag the actual API fields as well so the scripts pick it up, no?

Unfortunately there are no API fields here. This is related to this code:

https://github.com/envoyproxy/envoy/blob/8d1ad35aa724962f64f7535531e408c9a93d364c/source/extensions/filters/network/http_connection_manager/config.cc#L324

and 

https://github.com/envoyproxy/envoy/blob/8d1ad35aa724962f64f7535531e408c9a93d364c/source/server/listener_manager_impl.cc#L66

I agree with @derekargueta that we need to engage the stat/warn logic here, but it will need special casing. I was being lazy. Let me look into how to handle this. @alyssawilk any quick thoughts on how to one off engage the logic here since we aren't dealing with proto fields?

**Comment by mattklein123** [comment]
> @PiotrSikora for parity comments, since this relates to #5355

It doesn't actually relate to bind to port. See ^.

**Comment by alyssawilk** [comment]
Yeah, sorry, I was misreading the bit about the tcp proxy session and network filter config to think we were removing deprecated_v1 fields altogether

**Comment by mattklein123** [comment]
@alyssawilk I still want to address @derekargueta comment about engaging the stat/warn/fail by default behavior. Do you think it would be possible to actually guard these if checks with some type of fake runtime field check? I can look into it later today.

**Comment by mattklein123** [comment]
@derekargueta @alyssawilk updated PTAL

File: source/common/config/utility.cc [hunk 1]
```diff
@@ -287,5 +287,25 @@ void Utility::translateOpaqueConfig(const ProtobufWkt::Any& typed_config,
   }
 }
 
+bool Utility::allowDeprecatedV1Config(Runtime::Loader& runtime, const Json::Object& config) {
+  if (!config.getBoolean("deprecated_v1", false)) {
+    return false;
+  }
+
+  const char error[] =
```
Comment (derekargueta) [inline source/common/config/utility.cc]: nit `constexpr`

File: docs/root/intro/deprecated.rst [hunk 2]
```diff
@@ -21,6 +21,9 @@ Version 1.11.0 (Pending)
 * Use of :ref:`runtime <envoy_api_field_config.bootstrap.v2.Bootstrap.runtime>` in :ref:`Bootstrap
   <envoy_api_msg_config.bootstrap.v2.Bootstrap>`. Use :ref:`layered_runtime
   <envoy_api_field_config.bootstrap.v2.Bootstrap.layered_runtime>` instead.
+* Specifying "deprecated_v1: true" in HTTP and network filter configuration to allow loading JSON
```
Comment (alyssawilk) [inline docs/root/intro/deprecated.rst]: Do you think it's worth mentioning the runtime control either in the relnotes, the log message, or both?

Comment (mattklein123) [reply docs/root/intro/deprecated.rst]: Yeah good call let me do that.

**Comment by mattklein123** [comment]
@alyssawilk updated

**Pull request merged**